kolibri/plugins/learn/assets/src/views/content-card-carousel/index.vue

```diff
@@ -0,0 +1,300 @@
+<template>
+
+  <section class="content-carousel">
+
+    <div class="content-carousel-details">
+      <header v-if="header" class="content-carousel-details-header">
+        <h2> {{header}} </h2>
+        <span v-if="subheader"> {{subheader}} </span>
+      </header>
+    </div>
+
+    <div :style="widthOfCarousel" class="content-carousel-controls">
+      <div class="previous" @click="previousSet">
+        <ui-icon-button
+        class="previous-button"
+        v-show="!isFirstSet"
+        :disabled="isFirstSet"
+        icon="arrow_back"
+        size="large" />
+      </div>
+
+      <div class="next" @click="nextSet">
+        <ui-icon-button
+        class="next-button"
+        v-show="!isLastSet"
+        :disabled="isLastSet"
+        icon="arrow_forward"
+        size="large"/>
+      </div>
+    </div>
+
+    <transition-group
+      :style="widthOfCarousel"
+      class="content-carousel-set"
+      tag="div"
+      @leave="slide"
+      @before-enter="setStartPosition"
+      @enter="slide">
+
+      <div class="content-carousel-card"
+        v-for="(content, index) in contents"
+        v-if="isInThisSet(index)"
+        :style="positionCalc(index)"
+        :key="content.id">
+        <!-- uses props if scoped slot is unused -->
+          <slot
+            :title="content.title"
+            :thumbnail="content.thumnail"
+            :kind="content.kind"
+            :progress="content.progress"
+            :id="content.id">
+
+            <content-card
+            :title="content.title"
+            :thumbnail="content.thumbnail"
+            :kind="content.kind"
+            :progress="content.progress"
+            :link="genLink(content.id, content.kind)"/>
+          </slot>
+      </div>
+
+    </transition-group>
+
+  </section>
+
+</template>
+
+
+<script>
+
+  const responsiveElement = require('kolibri.coreVue.mixins.responsiveElement');
+  const validateLinkObject = require('kolibri.utils.validateLinkObject');
+
+  const contentCardWidth = 210;
+  const gutterWidth = 20;
+
+  module.exports = {
+    mixins: [responsiveElement],
+    $trNameSpace: 'contentCardCarousel',
+    $trs: {
+      viewAllButtonLabel: 'View all'
+    },
+    props: {
+      contents: {
+        type: Array,
+        required: true,
+      },
+      header: {
+        type: String,
+      },
+      subheader: {
+        type: String,
+      },
+      genLink: {
+        type: Function,
+        validator(value) {
+          return validateLinkObject(value(1, 'exercise'));
```

Comment:
Passes dummy variables into the `genLink()` function passed in as a prop, then passes the return value in `validateLinkObject`. Sloppy, but it's a simple way of testing whether the `genLink` prop does what's intended.